src/storage/store/engine.rs

```diff
@@ -203,13 +210,30 @@ impl ShardEngine {
         transactions
     }
 
+    fn incr(&self, key: &str) {
+        let key = format!("shard{}.{}", self.shard_id, key);
+        _ = self.metrics_client.incr(key.as_str())
+    }
+
+    fn count(&self, key: &str, value: u64) {
+        let key = format!("shard{}.{}", self.shard_id, key);
+        _ = self.metrics_client.count(key.as_str(), value)
+    }
+
+    fn gauge(&self, key: &str, value: u64) {
+        let key = format!("shard{}.{}", self.shard_id, key);
+        _ = self.metrics_client.gauge(key.as_str(), value)
+    }
```

Comment:
I think we should do this in a wrapper around the metrics client (similar to logging spans). 

Also, I think the shard indexes should be tags.